internal/codeConfiguration/task.ts

```diff
@@ -0,0 +1,48 @@
+import { execute } from "../execute";
+import { ParamSchema, JSParamValues, ParamKinds, ParamSchemaKind } from "./paramTypes";
+
+export type Params = Record<string, ParamKinds | ParamSchema>;
+
+export type ParamValues<PSchema extends Params> = {
```

Comment:
Let's use `T<...>` as a convention for types, to avoid naming collisions:

```suggestion
export type ParamValues<TParams extends Params> = {
```